diff --git a/node_modules/@react-native-voice/voice/android/src/main/java/com/wenkesj/voice/VoiceModule.java b/node_modules/@react-native-voice/voice/android/src/main/java/com/wenkesj/voice/VoiceModule.java
index f22833e..ded91fa 100644
--- a/node_modules/@react-native-voice/voice/android/src/main/java/com/wenkesj/voice/VoiceModule.java
+++ b/node_modules/@react-native-voice/voice/android/src/main/java/com/wenkesj/voice/VoiceModule.java
@@ -321,9 +321,11 @@ public class VoiceModule extends ReactContextBaseJavaModule implements Recogniti
   public void onPartialResults(Bundle results) {
     WritableArray arr = Arguments.createArray();
 
-    ArrayList<String> matches = results.getStringArrayList(SpeechRecognizer.RESULTS_RECOGNITION);
-    for (String result : matches) {
-      arr.pushString(result);
+    ArrayList<String> matches = results == null ? null : results.getStringArrayList(SpeechRecognizer.RESULTS_RECOGNITION);
+    if (matches != null) {
+      for (String result : matches) {
+        arr.pushString(result);
+      }
     }
 
     WritableMap event = Arguments.createMap();
@@ -344,9 +346,11 @@ public class VoiceModule extends ReactContextBaseJavaModule implements Recogniti
   public void onResults(Bundle results) {
     WritableArray arr = Arguments.createArray();
 
-    ArrayList<String> matches = results.getStringArrayList(SpeechRecognizer.RESULTS_RECOGNITION);
-    for (String result : matches) {
-      arr.pushString(result);
+    ArrayList<String> matches = results == null ? null : results.getStringArrayList(SpeechRecognizer.RESULTS_RECOGNITION);
+    if (matches != null) {
+      for (String result : matches) {
+        arr.pushString(result);
+      }
     }
 
     WritableMap event = Arguments.createMap();
